backend/src/BuiltinExecution/Libs/Int.fs

```diff
@@ -339,20 +356,40 @@ let fns : List<BuiltInFn> =
     { name = fn "parse" 0
       typeParams = []
       parameters = [ Param.make "s" TString "" ]
-      returnType = TypeReference.result TInt TString
+      returnType =
+        TypeReference.result
+          TInt
+          (TCustomType(
+            Ok(
+              FQName.Package
+                { owner = "Darklang"
+                  modules = [ "Stdlib"; "Int" ]
+                  name = TypeName.TypeName "IntParseError"
+                  version = 0 }
+            ),
+            []
+          ))
       description = "Returns the <type Int> value of a <type String>"
       fn =
         let resultOk = Dval.resultOk KTInt KTString
         let resultError = Dval.resultError KTInt KTString
         (function
         | _, _, [ DString s ] ->
-          try
-            s |> System.Convert.ToInt64 |> DInt |> resultOk |> Ply
-          with _e ->
-            $"Expected to parse String with only numbers, instead got \"{s}\""
-            |> DString
-            |> resultError
-            |> Ply
+          let trimmedString = s.Trim()
+          if
+            not (
+              System.Text.RegularExpressions.Regex.IsMatch(
+                trimmedString,
+                @"^[+-]?\d+$"
+              )
+            )
+          then
+            IntParseError.BadFormat |> IntParseError.toDT |> resultError |> Ply
+          else
+            match System.Int64.TryParse(s) with
+            | true, value -> value |> DInt |> resultOk |> Ply
+            | false, _ ->
+              IntParseError.OutOfRange |> IntParseError.toDT |> resultError |> Ply
```

Comment:
I was imagining this implementation to be different:
- keep the `try` as it was
- rather than one big blanket `with _e`, capture specific kinds of exceptions - see HttpClient implementation

Reply:
for example, the bit in httpClient.fs where we 
```fsharp
with
      | :? TaskCanceledException ->
```

Reply:
This is much better. Thanks!